### ISSUE TITLE
Templating behaves differently from Jinja2 documentation

### ISSUE THREAD

**Issue by rmbleeker** [ansible/ansible#67163]
##### SUMMARY
If this has been reported/discussed before I apologize but I couldn't find it.

The Jinja2 template designer documentation for the [IF statement](https://jinja.palletsprojects.com/en/2.10.x/templates/#if) clearly states:

> The if statement in Jinja is comparable with the Python if statement. In the simplest form, you can use it to test if a variable is defined, not empty and not false:

This "simplest form" is just using the plain variable in the IF statement without tests or filters. However Ansible throws an error when trying to deploy a template written in this way.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Template

##### ANSIBLE VERSION
```
ansible 2.9.1
ansible python module location = /usr/lib/python2.7/site-packages/ansible
executable location = /usr/bin/ansible
python version = 2.7.5 [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### OS / ENVIRONMENT
Red Hat 7.7

##### STEPS TO REPRODUCE
Example playbook:
```yaml
---
- name: deploy a template
  hosts: all
  tasks:
    - template:
        src: template.j2
        dest: /tmp/file
```
Example template:
```jinja2
This should be the only line in the file
{% if my_undefined_variable %}
variable is defined after all
{% endif %}
```

##### EXPECTED RESULTS
For Ansible to deploy the template as a file on the target host with only the first line, leaving out the second line because the IF statement result should be "False" when checking an undefined variable.
```ShellSession
$ ansible-playbook deploy_a_template.yml -l host

PLAY [deploy a template]

TASK [Gathering Facts]
ok: [host]

TASK [template]
--- before
+++ after: /home/ansible/.ansible/tmp/ansible-local-28324wVpu_s/tmp1Gcnme/template.j2
@@ -0,0 +1 @@
+This should be the only line in the file

changed: [host]

PLAY RECAP
host: ok=2 changed=1 unreachable=0 failed=0 skipped=0 rescued=0 ignored=0
```

##### ACTUAL RESULTS
Ansible throws an error about the variable being undefined
```ShellSession
$ ansible-playbook deploy_a_template.yml -l host

PLAY [deploy a template]

TASK [Gathering Facts]
ok: [host]

TASK [template]
fatal: [host]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'my_undefined_variable' is undefined"}

PLAY RECAP
host: ok=1 changed=0 unreachable=0 failed=1 skipped=0 rescued=0 ignored=0
```

**Comment by rmbleeker** [comment]
##### WORKAROUND
This does work if you formulate the IF statement in the template to explicitly check if the variable is undefined:
```jinja2
{% if my_undefined_variable is defined and my_undefined_variable %}
variable is defined after all
{% endif %}
```

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/58835 https://github.com/ansible/ansible/issues/56017
